main/sources/hear-this.ts

```diff
@@ -29,46 +42,146 @@ class HearThis implements ProjectSource {
-    const bookNames = sortInCanonicalOrder(getDirectories(project.fullPath));
-    project.books = bookNames
-      .map((bookName: string) => this.makeBook(name, bookName, directory))
-      .filter((book: Book) => book.chapters.length);
+  reloadProject(project: BKProject): BKProject {
+    const directory = path.parse(project.folderPath).dir;
+    const books = project.books.map((book: BKBook): BKBook => {
+      const chapters = book.chapters.map(
+        (chapter: BKChapter): BKChapter => this.makeChapter(project.name, book.name, chapter.name, directory, true)
+      );
+      return { name: book.name, chapters };
+    });
+    return { name: project.name, folderPath: project.folderPath, sourceType: project.sourceType, books };
+  }
+
+  private makeProject(projectName: string, directory: string): BKProject {
+    const project: BKProject = {
+      name: projectName,
+      folderPath: path.join(directory, projectName),
+      sourceType: this.SOURCE_TYPE,
+      books: [],
+    };
+    const bookNames = sortInCanonicalOrder(getDirectories(project.folderPath));
+    const projectBooks = bookNames
+      .map((bookName: string) => this.makeBook(projectName, bookName, directory))
+      .filter((book: BKBook) => book.chapters.length);
+    for (const book of projectBooks) {
+      project.books.push(book);
+    }
     return project;
   }
 
-  makeBook(projectName: string, name: string, directory: string): Book {
-    const book = new Book();
-    book.name = name;
-    const chapterNames = getDirectories(path.join(directory, projectName, name));
+  private makeBook(projectName: string, bookName: string, directory: string): BKBook {
+    const book: BKBook = {
+      name: bookName,
+      chapters: [],
+    };
+    const chapterNames = getDirectories(path.join(directory, projectName, bookName));
 
     const naturalSortChapterNames = chapterNames.sort(
       new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' }).compare
     );
 
-    book.chapters = naturalSortChapterNames
-      .map((chapterName: string) => this.makeChapter(projectName, name, chapterName, directory))
-      .filter((chapter: Chapter) => chapter.audioFiles.length > 0);
+    const bookChapters = naturalSortChapterNames
+      .map((chapterName: string) => this.makeChapter(projectName, bookName, chapterName, directory))
+      .filter((chapter: BKChapter) => chapter.audio.files.length);
+    for (const chapter of bookChapters) {
+      book.chapters.push(chapter);
+    }
     return book;
   }
 
-  makeChapter(projectName: string, bookName: string, name: string, directory: string): Chapter {
-    const chapter = new Chapter();
-    chapter.name = parseInt(name).toString();
-    const chapterFiles = fs.readdirSync(path.join(directory, projectName, bookName, name));
-    chapter.audioFiles = chapterFiles
-      .filter((file: string) => isValidAudioFile(file, DEFAULT_HEARTHIS_XML_FILE, fileFilters.audio[0].extensions))
```

Comment:
You have removed the use of the file filters. Do you know for certain that HearThis files will only be WAV and not ever MP3?

Reply:
I figured that HearThis would always generate audio with the same type of extension (lowercase .wav). But it's possible that a future update could change this so I have re-added the `checkValidAudioFile` function.